cmd/abapEnvironmentRunATCCheck.go

```diff
@@ -0,0 +1,354 @@
+	var ATCConfig ATCconfig
+	if err == nil {
+		var result []byte
+		result, err = yaml.YAMLToJSON(yamlFile)
+		json.Unmarshal(result, &ATCConfig)
+	}
+
+	var packageString string
+	var softwareComponentString string
+	if err == nil {
+		packageString, softwareComponentString, err = buildATCCheckBody(ATCConfig)
+	}
+
+	//Trigger ATC run
+	var resp *http.Response
+	var bodyString = `<?xml version="1.0" encoding="UTF-8"?><atc:runparameters xmlns:atc="http://www.sap.com/adt/atc" xmlns:obj="http://www.sap.com/adt/objectset"><obj:objectSet>` + softwareComponentString + packageString + `</obj:objectSet></atc:runparameters>`
+	var body = []byte(bodyString)
+
+	if err == nil {
+		details.URL = abapEndpoint + "/sap/bc/adt/api/atc/runs?clientWait=false"
+		resp, err = runATC("POST", details, body, &client)
+	}
+
+	var location string
+	//Poll ATC run
+	if err == nil {
+		location = resp.Header.Get("Location")
+		details.URL = abapEndpoint + location
+		location, err = pollATCRun(details, body, &client)
+	}
+
+	if err == nil {
+		details.URL = abapEndpoint + location
+		resp, err = getResultATCRun("GET", details, nil, &client)
+	}
+
+	//Parse response
+	if err == nil {
+		body, err = ioutil.ReadAll(resp.Body)
+	}
+	if err == nil {
+		defer resp.Body.Close()
+		err = parseATCResult(body)
+	}
+
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+
+	log.Entry().Info("ATC run completed succesfully. The respective run results are listed above.")
+}
+
+func buildATCCheckBody(ATCConfig ATCconfig) (string, string, error) {
+	if len(ATCConfig.Objects.Package) == 0 && len(ATCConfig.Objects.SoftwareComponent) == 0 {
+		return "", "", fmt.Errorf("Error while parsing ATC run config. Please provide the packages and/or the software components to be checked! %w", errors.New("No Package or Software Component specified. Please provide either one or both of them"))
+	}
+
+	var packageString string
+	var softwareComponentString string
+
+	//Build Package XML body
+	if len(ATCConfig.Objects.Package) != 0 {
+		packageString += "<obj:packages>"
+		for _, s := range ATCConfig.Objects.Package {
+			packageString += `<obj:package value="` + s.Name + `" includeSubpackages="` + strconv.FormatBool(s.IncludeSubpackages) + `"/>`
+		}
+		packageString += "</obj:packages>"
+	}
+
+	//Build SC XML body
+	if len(ATCConfig.Objects.SoftwareComponent) != 0 {
+		softwareComponentString += "<obj:softwarecomponents>"
+		for _, s := range ATCConfig.Objects.SoftwareComponent {
+			softwareComponentString += `<obj:softwarecomponent value="` + s.Name + `"/>`
+		}
+		softwareComponentString += "</obj:softwarecomponents>"
+	}
+	return packageString, softwareComponentString, nil
+}
+
+func parseATCResult(body []byte) error {
+	if len(body) == 0 {
+		return fmt.Errorf("Parsing ATC result failed: %w", errors.New("Body is empty, can't parse empty body"))
+	}
+	parsedXML := new(Result)
+	xml.Unmarshal([]byte(body), &parsedXML)
+	err := ioutil.WriteFile("result.xml", body, 0644)
```

Comment:
Is this step creating a file that should be pinned to the Jenkins Job execution? Then consider providing it via the [step results](https://github.com/SAP/jenkins-library/blob/master/pkg/piperutils/stepResults.go) (like in [sonar](https://github.com/SAP/jenkins-library/blob/master/cmd/sonarExecuteScan.go#L118) or [checkmarx](https://github.com/SAP/jenkins-library/blob/master/cmd/checkmarxExecuteScan.go#L156)).

Reply:
The `piperutils.PersistReportsAndLinks()` function will produce two JSON files, whereas this step needs to output the ATC results in a XML file.
Took the `piperenv.SetParameter()` function. Hope that does the job.

Reply:
The `piperutils.PersistReportsAndLinks` will produce JSON files that contain name and link/filename for reports to be published on the Jenkins jobs. The JSON files are handles afterwards in `piperExecuteBin` automatically.

Reply:
Got it. When testing locally, I was questioning why these files appear. After running it inside a Jenkinsfile with the Groovy Script everything works and the Results are getting pinned to the Jenkins Job execution.
Thanks for the hint!